File: junechallenge2019/PROXYC.py
import math
t = int(input())

while t > 0:
    t-=1
    size = int(input())
    s = input()
    quant = math.ceil(size*3/4.0)
    presencas = int(0)
    possiveis = int(0)
    for d in s:
        if d == 'P': presencas+=1

    for i in range(2, size - 2):
        if s[i] == 'A':
            if (s[i-1] == 'P' or s[i-2] == 'P') and ( s[i+1] == 'P' or s[i+2] == 'P'):
                possiveis+=1

    if quant - presencas <= possiveis:
        print(str(quant-presencas))
    else:
        print("-1")

File: junechallenge2019/PROXYC.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
  int t; cin >> t;
  while(t--){
    string s;
    int size; cin >> size;
    cin >> s;

    int presencas = 0;
    int possiveis = 0;

    for(auto c: s){
      if (c == 'P') presencas++;
    }

    for(int i = 2; i < size - 2; i++){
      if(s[i] == 'A' &&
        ( (s[i-1] == 'P') || (s[i-2] == 'P') )
        &&
        ( (s[i+1] == 'P') || (s[i+2] == 'P') )
      )
      possiveis++;

    }

    if(presencas*4 >= size*3) cout << "0" << endl;
    else if( size*3 > (presencas+possiveis)*4) cout << "-1" << endl;
    else cout << ceil(3.00*size/4.0) - presencas << endl;

  }
}
